daemon/containerd/image.go

```diff
@@ -197,7 +201,44 @@ func (i *ImageService) resolveImage(ctx context.Context, refOrID string) (img co
 		}
 		return containerdimages.Image{}, errdefs.NotFound(errors.New("id not found"))
 	}
+	if platform != nil {
+		cs := i.client.ContentStore()
+		comparer := platforms.Only(*platform)
+		found := false
+
+		containerdimages.Walk(ctx,
+			containerdimages.Handlers(containerdimages.HandlerFunc(func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
+				if desc.Platform != nil && comparer.Match(*desc.Platform) {
+					found = true
+					return nil, containerdimages.ErrSkipDesc
+				}
+
+				switch desc.MediaType {
+				case containerdimages.MediaTypeDockerSchema2Config, ocispec.MediaTypeImageConfig:
+					p, err := content.ReadBlob(ctx, cs, desc)
+					if err != nil {
+						return nil, err
+					}
+
+					var image ocispec.Image
+					if err := json.Unmarshal(p, &image); err != nil {
+						return nil, err
+					}
 
+					pf := platforms.Normalize(ocispec.Platform{OS: image.OS, Architecture: image.Architecture})
+					if comparer.Match(pf) {
+						found = true
+						return nil, containerdimages.ErrSkipDesc
+					}
+				}
+				return nil, nil
+			}), containerdimages.ChildrenHandler(cs)),
+			img.Target)
```

Comment:
Also, there is a way to use c8d API here which I think would have the same effect, if you'd prefer having less code 😄 

```suggestion
		found := false
		imgPlatforms, err := containerdimages.Platforms(ctx, cs, img.Target)
		if err != nil {
			return containerdimages.Image{}, err
		}
		for _, p := range imgPlatforms {
			if comparer.Match(p) {
				found = true
				break
			}
		}

```

Reply:
Bad I didn't noticed this one, and basically re-implemented the same 😅